internal/config/profile_test.go

```diff
@@ -0,0 +1,304 @@
+	atlasProfile   = "atlas"
+	newProfileName = "newProfileName"
+)
+
+var testConfigDir = "/test_dir"
+var fs afero.Fs
+var filename = fmt.Sprintf("%v/%v.toml", testConfigDir, ToolName)
+
+func TestMain(m *testing.M) {
+	fs = afero.NewMemMapFs()
+
+	viper.SetFs(fs)
+	SetConfigPath(testConfigDir)
+
+	code := m.Run()
+	os.Exit(code)
+}
+
+func createProfile(profileContents string) (*profile, error) {
+	err := afero.WriteFile(fs, filename, []byte(profileContents), 0600)
+	if err != nil {
+		return nil, err
+	}
+
+	return &profile{
+		name:      "default",
+		configDir: testConfigDir,
+		fs:        fs,
+	}, nil
+}
+
+func createProfileWithJustDefaultUser() (*profile, error) {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithOneNonDefaultUser() (*profile, error) {
+	const contents = `
+[atlas]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithOneDefaultUserOneNonDefault() (*profile, error) {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+  profile_id = "5cac6a2179358edabd12b572"
+
+[atlas]
+  base_url = "http://atlas.com/"
+  ops_manager_skip_verify = "false"
+  org_id = "5cac6a2179358edabd12b572"
+`
+	return createProfile(contents)
+}
+
+func createProfileWithFullDescription() (*profile, error) {
+	const contents = `
+[default]
+  base_url = "http://base_url.com/"
+  ops_manager_url = "http://om_url.com/"
+  ops_manager_ca_certificate = "/path/to/certificate"
+  ops_manager_skip_verify = "false"
+  public_api_key = "some_public_key"
+  private_api_key = "some_private_key"
+  org_id = "5cac6a2179358edabd12b572"
+  profile_id = "5cac6a2179358edabd12b572"
+  service = "cloud"
+`
+	return createProfile(contents)
+}
+
+func TestProfileList(t *testing.T) {
+	profile, err := createProfileWithOneDefaultUserOneNonDefault()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if err := profile.Load(false); err != nil {
+		t.Error(err)
+	}
+
+	availableProfiles := profile.List()
+	assert.Equal(t, 2, len(availableProfiles), "expected to find 2 profiles")
+}
+
+func TestProfileDescribeFullProfile(t *testing.T) {
+	profile, err := createProfileWithFullDescription()
+	if err != nil {
+		t.Error(err)
+	}
+
+	if err := profile.Load(false); err != nil {
+		t.Error(err)
+	}
+
+	desc := profile.GetConfigDescription()
+	assert.Equal(t, "default", Name())
+	assert.Equal(t, 9, len(desc))
+	assert.Equal(t, "cloud", Service())
+	assert.Equal(t, "some_public_key", PublicAPIKey())
+	assert.Equal(t, "some_private_key", PrivateAPIKey())
+	assert.Equal(t, "http://om_url.com/", OpsManagerURL())
+	assert.Equal(t, "/path/to/certificate", OpsManagerCACertificate())
+	assert.Equal(t, "false", OpsManagerSkipVerify())
```

Comment:
reading the docs of assert they recommend that when you do multiple assertions is better to do 
`a : = assert.New()` then all your `.Equal` should be against `a` like `a.Equal` (I'm using a instead of assert as they docs as our linter will tell you you're shadowing an import that way)